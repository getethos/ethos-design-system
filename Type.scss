/* @getethos/design-system/Type.scss
   ========================================================================== */

// Typefaces

:root {
  --system-sans-font-stack: -apple-system, BlinkMacSystemFont, Roboto,
    'Segoe UI', 'Helvetica Neue', Helvetica, Arial, sans-serif;
  --UntitledSans-font-stack: 'UntitledSans', var(--system-sans-font-stack);
  --Americane-font-stack: 'Americane', var(--UntitledSans-font-stack);
}

@font-face {
  font-family: 'Americane';
  font-weight: 900;
  src: url('/fonts/Americane-900-Black.woff2') format('woff2'),
    url('/fonts/Americane-900-Black.woff') format('woff');
}

@font-face {
  font-family: 'UntitledSans';
  font-weight: 400;
  src: url('/fonts/UntitledSans-400-Regular.woff2') format('woff2'),
    url('/fonts/UntitledSans-400-Regular.woff') format('woff');
}

@font-face {
  font-family: 'UntitledSans';
  font-weight: 500;
  src: url('/fonts/UntitledSans-500-Medium.woff2') format('woff2'),
    url('/fonts/UntitledSans-500-Medium.woff') format('woff');
}

// Sizes

:root {
  --XXXL-font-size: 48px;
  --XXXL-mobile-font-size: 36px;
  --XXL-font-size: 32px;
  --XXL-mobile-font-size: 28px;
  --XL-font-size: 22px;
  --L-font-size: 18px;
  --M-font-size: 16px;
  --S-font-size: 14px;
  --XS-font-size: 12px;

  --Heading-line-height: 1.25;
  --Text-line-height: 1.6;
}

.XXXL {
  font-size: var(--XXXL-mobile-font-size);
  @media (min-width: 900px) {
    font-size: var(--XXXL-font-size);
  }
}

.XXL {
  font-size: var(--XXL-mobile-font-size);
  @media (min-width: 900px) {
    font-size: var(--XXL-font-size);
  }
}

.XL {
  font-size: var(--XL-font-size);
}

.L {
  font-size: var(--L-font-size);
}

.M {
  font-size: var(--M-font-size);
}

.S {
  font-size: var(--S-font-size);
}

.XS {
  font-size: var(--XS-font-size);
}

// Global defaults

@mixin global-type-defaults {
  font-family: var(--UntitledSans-font-stack);
  line-height: var(--Text-line-height);
  color: var(--Black--translucent);
}

html {
  font-size: var(--M-font-size);
}

body {
  font-size: var(--M-font-size);
  @include global-type-defaults;
}

// Headings

h1,
h2,
h3,
h4,
h5,
h6,
.Heading {
  line-height: var(--Heading-line-height);
  margin: 0;
}

.Heading.UntitledSans,
.Heading.UntitledSans.Medium500 {
  font-family: var(--UntitledSans-font-stack);
  font-weight: 500;
  margin: 0;
}

.Heading.UntitledSans.Regular400 {
  font-weight: 400;
}

.Heading.Americane,
.Heading.Americane.Black900 {
  font-family: var(--Americane-font-stack);
  font-weight: 900;
}

// Text

.Text {
  @include global-type-defaults;
  display: block;
}

.Text b,
.Text strong,
.Text.Medium500 {
  font-weight: 500;
}

// Colors

.Heading,
.Text {
  &.BrandYellow {
    color: var(--BrandYellow--default);
  }

  &.BrandBlue {
    color: var(--BrandBlue--default);
  }

  &.Black {
    color: var(--Black--translucent);
  }

  &.GrayDarkest {
    color: var(--GrayDarkest--translucent);
  }

  &.GrayDark {
    color: var(--GrayDark--translucent);
  }

  &.GrayLight {
    color: var(--GrayLight--translucent);
  }

  &.GrayLightest {
    color: var(--GrayLightest--translucent);
  }

  &.White {
    color: var(--White);
  }

  &.Sky {
    color: var(--Sky);
  }
}

// Aliases - these shouldn't need custom CSS ideally, but do atm:

.Eyebrow {
  text-transform: uppercase;
  letter-spacing: 0.12em;
}
