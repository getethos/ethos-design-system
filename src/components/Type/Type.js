import React from 'react'
import PropTypes from 'prop-types'

import useIncludes from '../../hooks/useIncludes.js'
import useInvalid from '../../hooks/useInvalid.js'
import { COLORS } from '../Colors'

/* @getethos/design-system/Type.js

   Legend:

   - `Type` is a private component that returns an element with CSS classes.
   - `Type.scss` (not imported here) implements those CSS classes.
   - `TypeFoundry` is a HOC that creates public components with correct props.
   - `<Caption.Medium500>`, etc. are the Design-approved public components.
   ========================================================================== */

/**
 * `Type` is a private component that returns an element with CSS classes.
 *
 * Note that it is ignorant of which combinations Design considers legal or
 * illegal. Legal combinations must be declared via the enumerated exports at
 * the end of this file.
 *
 * The public props (Type.PUBLIC_PROPS) may be passed in downstream.
 * Other props may only be specified in this file.
 *
 * @param  {String}  props.children       The text to display
 * @param  {Boolean} props.centered       Whether to text-align: center
 * @param  {Boolean} props.allCaps        Whether to text-transform to: uppercase
 * @param  {String}  props.element        Override the default <element>
 * @param  {String}  props.subtype        (private) e.g. Caption vs. TitleSmall
 * @param  {String}  props.typeface       (private) Typeface
 * @param  {String}  props.weight         (private) Typeface weight
 */
export const Type = ({
  children,
  centered,
  allCaps,
  color,
  element,
  subtype,
  typeface,
  weight,
  ...rest
}) => {
  // Verify that color, subtype, typeface, and weight were valid enum values
  const [isValidColor] = useIncludes(Type.COLORS)
  isValidColor(color)
  const [isValidSubtype] = useIncludes(Type.SUBTYPES)
  isValidSubtype(subtype)
  const [isValidTypeface] = useIncludes(Type.TYPEFACES)
  isValidTypeface(typeface)
  const [isValidWeight] = useIncludes(Type.WEIGHTS)
  isValidWeight(weight)

  const WHITELISTED_PROPS = ['htmlFor', 'data-tid']
  const [, isWhiteListedProp] = useIncludes(WHITELISTED_PROPS)
  isWhiteListedProp(rest)

  // Verify that no invalid props were supplied
  const [includesInvalid] = useInvalid(Object.keys(Type.PUBLIC_PROPS))
  includesInvalid(rest)

  // Generate list of css classes
  const classNames = [subtype, typeface, weight, color]
  if (centered) classNames.push('Centered')
  if (allCaps) classNames.push('AllCaps')

  // Defaults to div, but can be overridden
  const Element = element || 'div'

  const allowedProps = Object.keys(rest).reduce((acc, key) => {
    acc[key] = rest[key]
    return acc
  }, {})

  return (
    <Element className={classNames.join(' ')} {...allowedProps}>
      {children}
    </Element>
  )
}

Type.SUBTYPES = {
  CAPTION: 'Caption', // smallest
  FOOTNOTE: 'Footnote',
  BODY: 'Body', // default
  TITLE_SMALL: 'TitleSmall',
  TITLE_MEDIUM: 'TitleMedium',
  TITLE_LARGE: 'TitleLarge',
  TITLE_XLARGE: 'TitleXLarge',
  TITLE_XXLARGE: 'TitleXXLarge',
}

Type.TYPEFACES = {
  THEINHARDT: 'Theinhardt', // sans
  CAMBON: 'Cambon', // serif
}

Type.WEIGHTS = {
  // Possibly we shouldn't lump these all together given they vary per typeface.
  REGULAR_400: 'Regular400',
  MEDIUM_500: 'Medium500',
  BOOK_500: 'Book500',
  DEMI_600: 'Demi600',
}

Type.COLORS = {
  // Brand
  BRAND_FOREST: COLORS.BRAND_FOREST,
  BRAND_SALAMANDER: COLORS.BRAND_SALAMANDER,

  // Grayscale
  GRAY_PRIMARY: COLORS.GRAY_PRIMARY,
  GRAY_SECONDARY: COLORS.GRAY_SECONDARY,
  WHITE: COLORS.WHITE,
}

Type.ELEMENTS = {
  H1: 'h1',
  H2: 'h2',
  H3: 'h3',
  H4: 'h4',
  H5: 'h5',
  H6: 'h6',
  DIV: 'div',
  LI: 'li',
  SPAN: 'span',
  LABEL: 'label',
}

Type.PUBLIC_PROPS = {
  children: PropTypes.node,
  centered: PropTypes.bool,
  allCaps: PropTypes.bool,
  color: PropTypes.oneOf(Object.values(Type.COLORS)),
  element: PropTypes.oneOf(Object.values(Type.ELEMENTS)),
  htmlFor: PropTypes.string,
  'data-tid': PropTypes.string,
}

Type.propTypes = {
  ...Type.PUBLIC_PROPS,
  typeface: PropTypes.oneOf(Object.values(Type.TYPEFACES)),
  weight: PropTypes.oneOf(Object.values(Type.WEIGHTS)),
}

Type.defaultProps = {
  color: Type.COLORS.GRAY_PRIMARY,
}

// function TypeFoundry(privateProps) {
export const TypeFoundry = (privateProps) => {
  function throwIllegalProp(prop) {
    const isIllegal = !Object.keys(Type.PUBLIC_PROPS).includes(prop)
    if (isIllegal) throw new TypeError(`Illegal prop '${prop}'`)
  }

  // TODO: figure out if downstream or upstream is the correct nomenclature here
  const PublicTypeComponent = (downstreamProps) => {
    Object.keys(downstreamProps).forEach(throwIllegalProp)
    return <Type {...downstreamProps} {...privateProps} />
  }

  return PublicTypeComponent
}
