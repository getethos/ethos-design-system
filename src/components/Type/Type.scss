/* @getethos/design-system/Type.scss
   ========================================================================== */

@import '../Media/Media.scss';

// Typefaces

:root {
  --system-sans-font-stack: -apple-system, BlinkMacSystemFont, 'Helvetica Neue',
    Helvetica, Arial, sans-serif;
  --system-serif-font-stack: Georgia, 'Times New Roman', serif;
  --Theinhardt-font-stack: Theinhardt, var(--system-sans-font-stack);
  --Cambon-font-stack: Cambon, var(--system-serif-font-stack);
}

.Theinhardt {
  font-family: var(--Theinhardt-font-stack);
}

.Cambon {
  font-family: var(--Cambon-font-stack);
}

// Global defaults

body {
  font-family: var(--Theinhardt-font-stack);
  color: var(--GrayPrimary--translucent);
  word-break: break-word;
}

b,
strong {
  font-weight: 500;
}

// Type utilities (colors, weights, alignment)

.Caption,
.Footnote,
.Body,
.TitleSmall,
.TitleMedium,
.TitleLarge,
.TitleXLarge,
.TitleXXLarge {
  &.BrandForest {
    color: var(--BrandForest);
  }

  &.GraySecondary {
    color: var(--GraySecondary--translucent);
  }

  &.Regular400 {
    font-weight: 400;
  }

  &.Book500,
  &.Medium500 {
    font-weight: 500;
  }

  &.Demi600 {
    font-weight: 600;
  }

  &.Centered {
    text-align: center;
  }

  &.AllCaps {
    text-transform: uppercase;
  }
}

// Subtypes

.Caption {
  @include for-phone-only {
    $font-size: 12;
    $line-height: 15;
    font-size: $font-size * 1px;
    line-height: $line-height / $font-size;
  }

  @include for-tablet-and-up {
    $font-size: 13.5;
    $line-height: 16;
    font-size: $font-size * 1px;
    line-height: $line-height / $font-size;
  }
}

.Footnote {
  @include for-phone-only {
    $font-size: 15;
    $line-height: 18;
    font-size: $font-size * 1px;
    line-height: $line-height / $font-size;
  }

  @include for-tablet-and-up {
    $font-size: 15.5;
    $line-height: 20;
    font-size: $font-size * 1px;
    line-height: $line-height / $font-size;
  }
}

.Body {
  $font-size: 19;
  $line-height: 24;
  font-size: $font-size * 1px;
  line-height: $line-height / $font-size;
}

h1,
h2,
h3,
h4,
h5,
h6,
.TitleSmall,
.TitleMedium,
.TitleLarge,
.TitleXLarge,
.TitleXXLarge {
  margin: 0;
}

.TitleSmall {
  &.Theinhardt {
    @include for-phone-only {
      $font-size: 21.75;
      $line-height: 26;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 24;
      $line-height: 28;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 27;
      $line-height: 32;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }

  &.Cambon {
    @include for-phone-only {
      $font-size: 21.75;
      $line-height: 26;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 24;
      $line-height: 28;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 27;
      $line-height: 32;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }
}

.TitleMedium {
  &.Theinhardt {
    @include for-phone-only {
      $font-size: 23.5;
      $line-height: 28;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 28;
      $line-height: 32;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 35.5;
      $line-height: 40;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }

  &.Cambon {
    @include for-phone-only {
      $font-size: 27.5;
      $line-height: 32;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 32;
      $line-height: 36;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 33;
      $line-height: 40;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }
}

.TitleLarge {
  &.Theinhardt {
    @include for-phone-only {
      $font-size: 28;
      $line-height: 32;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 38;
      $line-height: 40;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 45;
      $line-height: 48;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }

  &.Cambon {
    @include for-phone-only {
      $font-size: 32;
      $line-height: 36;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 40;
      $line-height: 46;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 43.5;
      $line-height: 49;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }
}

.TitleXLarge {
  &.Theinhardt {
    @include for-phone-only {
      $font-size: 33;
      $line-height: 36;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 59;
      $line-height: 64;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 69;
      $line-height: 72;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }

  &.Cambon {
    @include for-phone-only {
      $font-size: 36.5;
      $line-height: 40;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 51;
      $line-height: 56;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 69.5;
      $line-height: 72;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }
}

.TitleXXLarge {
  &.Theinhardt {
    @include for-phone-only {
      $font-size: 44;
      $line-height: 41.5;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-only {
      $font-size: 76;
      $line-height: 76;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-laptop-only {
      $font-size: 78;
      $line-height: 80;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 92;
      $line-height: 92;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }

  &.Cambon {
    @include for-phone-only {
      $font-size: 41.5;
      $line-height: 44;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-tablet-and-laptop {
      $font-size: 71;
      $line-height: 76;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }

    @include for-desktop-only {
      $font-size: 90.5;
      $line-height: 96;
      font-size: $font-size * 1px;
      line-height: $line-height / $font-size;
    }
  }
}

// Links â€“ WIP. A link could mean a number of things for a given tech stack.

a,
.Link {
  &,
  &:active,
  &:hover {
    text-decoration: none;
  }

  & {
    color: var(--GrayPrimary--translucent);
  }
}

// Subtype components of Type that wrap an anchor tag
// This is a separate way to implement anchor tags than the actual Link (WIP) component
// Example: <Body><a>A link</a><Body>
// @todo :visited state ?
.Caption,
.Footnote,
.Body,
.TitleSmall,
.TitleMedium,
.TitleLarge,
.TitleXLarge,
.TitleXXLarge {
  a {
    &,
    &:active,
    &:hover {
      color: inherit;
      cursor: pointer;
      text-decoration: underline;
    }
  }
}

.Link {
  &.Inherit {
    color: inherit;
  }

  &.Navlink {
    color: var(--GrayPrimary--translucent);
    text-decoration: none;
    padding: var(--Space-4) 0;
    border-bottom: 2px solid transparent;

    // transition: var(--transition-default);
    // transition-property: none;

    &:active,
    &:hover {
      // &[aria-current='page'] // if we need to underline the current page later
      border-bottom-color: var(--GrayPrimary--translucent);
    }
  }
}
