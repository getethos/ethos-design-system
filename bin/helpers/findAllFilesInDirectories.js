const findAllFilesWithExt = require('./findAllFilesWithExt')

module.exports = function findAllFilesInDirectory(fileDirectories, extension) {
  return fileDirectories.reduce((memo, directory) => {
    return memo.concat(findAllFilesWithExt(directory, extension))
  }, [])
}
