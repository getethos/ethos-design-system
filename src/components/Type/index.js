export { Caption } from './Caption.js'
export { Footnote } from './Footnote.js'
export { Body } from './Body.js'
export { TitleSmall } from './TitleSmall.js'
export { TitleMedium } from './TitleMedium.js'
export { TitleLarge } from './TitleLarge.js'
export { TitleXLarge } from './TitleXLarge.js'
export { TitleXXLarge } from './TitleXXLarge.js'
export { Link } from './Link.js'
