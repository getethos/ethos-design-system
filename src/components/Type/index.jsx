export { Caption } from './Caption'
export { Footnote } from './Footnote'
export { Body } from './Body'
export { TitleSmall } from './TitleSmall'
export { TitleMedium } from './TitleMedium'
export { TitleLarge } from './TitleLarge'
export { TitleXLarge } from './TitleXLarge'
export { TitleXXLarge } from './TitleXXLarge'
export { Link } from './Link'
