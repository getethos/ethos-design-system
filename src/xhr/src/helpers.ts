import XhrError from './XhrError'

export interface KeyValMap {
  [key: string]: any
}

export interface XhrComponentState {
  status?: string
  error?: XhrError
  successMessage?: string
}

export type FormEvent = React.FormEvent<HTMLFormElement>

