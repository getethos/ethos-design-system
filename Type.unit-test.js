import React from 'react'

import { deepSnapshot } from 'test/helpers/snapshot'

import { Heading, Text } from './Type'

describe('Type.js', () => {
  describe('{ Heading }', () => {
    Object.keys(Heading.SIZES).forEach((Size) => {
      Object.keys(Heading[Size]).forEach((Style) => {
        describe(`<Heading.${Size}.${Style}>`, () => {
          const Component = Heading[Size][Style]
          it('matches snapshot', () => {
            expect(
              deepSnapshot(<Component>Lorem ipsum</Component>)
            ).toMatchSnapshot()
          })
        })
      })
    })
  })

  describe('{ Text }', () => {
    Object.keys(Text.SIZES).forEach((Size) => {
      Object.keys(Text[Size]).forEach((Style) => {
        describe(`<Text.${Size}.${Style}>`, () => {
          const Component = Text[Size][Style]
          it('matches snapshot', () => {
            expect(
              deepSnapshot(<Component>Lorem ipsum dolor sit amet</Component>)
            ).toMatchSnapshot()
          })
        })
      })
    })
  })
})
