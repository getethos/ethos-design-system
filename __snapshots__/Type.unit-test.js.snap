// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Type.js { Heading } <Heading.L.Medium> matches snapshot 1`] = `
<h4
  className="Heading L UntitledSans Medium500 Black"
>
  Lorem ipsum
</h4>
`;

exports[`Type.js { Heading } <Heading.L.Regular> matches snapshot 1`] = `
<h4
  className="Heading L UntitledSans Regular400 Black"
>
  Lorem ipsum
</h4>
`;

exports[`Type.js { Heading } <Heading.M.Medium> matches snapshot 1`] = `
<h5
  className="Heading M UntitledSans Medium500 Black"
>
  Lorem ipsum
</h5>
`;

exports[`Type.js { Heading } <Heading.M.Regular> matches snapshot 1`] = `
<h5
  className="Heading M UntitledSans Regular400 Black"
>
  Lorem ipsum
</h5>
`;

exports[`Type.js { Heading } <Heading.S.Medium> matches snapshot 1`] = `
<h6
  className="Heading S UntitledSans Medium500 Black"
>
  Lorem ipsum
</h6>
`;

exports[`Type.js { Heading } <Heading.S.Regular> matches snapshot 1`] = `
<h6
  className="Heading S UntitledSans Regular400 Black"
>
  Lorem ipsum
</h6>
`;

exports[`Type.js { Heading } <Heading.XL.Medium> matches snapshot 1`] = `
<h3
  className="Heading XL UntitledSans Medium500 Black"
>
  Lorem ipsum
</h3>
`;

exports[`Type.js { Heading } <Heading.XL.Regular> matches snapshot 1`] = `
<h3
  className="Heading XL UntitledSans Regular400 Black"
>
  Lorem ipsum
</h3>
`;

exports[`Type.js { Heading } <Heading.XS.Medium> matches snapshot 1`] = `
<h6
  className="Heading XS UntitledSans Medium500 Black"
>
  Lorem ipsum
</h6>
`;

exports[`Type.js { Heading } <Heading.XS.Regular> matches snapshot 1`] = `
<h6
  className="Heading XS UntitledSans Regular400 Black"
>
  Lorem ipsum
</h6>
`;

exports[`Type.js { Heading } <Heading.XXL.Americane> matches snapshot 1`] = `
<h2
  className="Heading XXL Americane Black900 Black"
>
  Lorem ipsum
</h2>
`;

exports[`Type.js { Heading } <Heading.XXL.Medium> matches snapshot 1`] = `
<h2
  className="Heading XXL UntitledSans Medium500 Black"
>
  Lorem ipsum
</h2>
`;

exports[`Type.js { Heading } <Heading.XXL.Regular> matches snapshot 1`] = `
<h2
  className="Heading XXL UntitledSans Regular400 Black"
>
  Lorem ipsum
</h2>
`;

exports[`Type.js { Heading } <Heading.XXXL.Americane> matches snapshot 1`] = `
<h1
  className="Heading XXXL Americane Black900 Black"
>
  Lorem ipsum
</h1>
`;

exports[`Type.js { Heading } <Heading.XXXL.Medium> matches snapshot 1`] = `
<h1
  className="Heading XXXL UntitledSans Medium500 Black"
>
  Lorem ipsum
</h1>
`;

exports[`Type.js { Heading } <Heading.XXXL.Regular> matches snapshot 1`] = `
<h1
  className="Heading XXXL UntitledSans Regular400 Black"
>
  Lorem ipsum
</h1>
`;

exports[`Type.js { Text } <Text.L.Medium> matches snapshot 1`] = `
<div
  className="Text L UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.L.Regular> matches snapshot 1`] = `
<div
  className="Text L UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.M.Medium> matches snapshot 1`] = `
<div
  className="Text M UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.M.Regular> matches snapshot 1`] = `
<div
  className="Text M UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.S.Medium> matches snapshot 1`] = `
<div
  className="Text S UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.S.Regular> matches snapshot 1`] = `
<div
  className="Text S UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XL.Medium> matches snapshot 1`] = `
<div
  className="Text XL UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XL.Regular> matches snapshot 1`] = `
<div
  className="Text XL UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XS.Medium> matches snapshot 1`] = `
<div
  className="Text XS UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XS.Regular> matches snapshot 1`] = `
<div
  className="Text XS UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XXL.Medium> matches snapshot 1`] = `
<div
  className="Text XXL UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XXL.Regular> matches snapshot 1`] = `
<div
  className="Text XXL UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XXXL.Medium> matches snapshot 1`] = `
<div
  className="Text XXXL UntitledSans Medium500 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;

exports[`Type.js { Text } <Text.XXXL.Regular> matches snapshot 1`] = `
<div
  className="Text XXXL UntitledSans Regular400 Black"
>
  Lorem ipsum dolor sit amet
</div>
`;
